GVFS/GVFS.UnitTests/Mock/FileSystem/MockFileSystem.cs

```diff
@@ -38,7 +38,7 @@ public MockFileSystem(MockDirectory rootDirectory)
         /// </summary>
         public bool DeleteNonExistentFileThrowsException { get; set; }
 
-        public override void RecursiveDelete(string path)
+        public override void DeleteDirectory(string path, bool recursive = true)
         {
             this.RootDirectory.DeleteDirectory(path);

```

Comment:
Since you're in here, could you updating this to throw if `recursive` is false (since we don't currently have code to mock that out).  I am pretty sure we're not hitting that code path in the unit tests.

Something like:

```
public override void DeleteDirectory(string path, bool recursive = true)
{
    if (!recursive)
    {
        throw new NotImplementedException();
    }

    this.RootDirectory.DeleteDirectory(path);
}
```